src/main/java/com/commercetools/sync/services/impl/ChannelServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.services.ChannelService;
+import io.sphere.sdk.channels.Channel;
+import io.sphere.sdk.channels.ChannelDraft;
+import io.sphere.sdk.channels.ChannelDraftBuilder;
+import io.sphere.sdk.channels.ChannelRole;
+import io.sphere.sdk.channels.commands.ChannelCreateCommand;
+import io.sphere.sdk.channels.queries.ChannelQuery;
+import io.sphere.sdk.channels.queries.ChannelQueryBuilder;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.queries.QueryExecutionUtils;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+public final class ChannelServiceImpl implements ChannelService {
```

Comment:
I see one more possible issue with the cache. As you said elsewhere:

> A very common use case of this library is streaming data from an XML or CSV file, and calling the sync(drafts) on each batch concurrently.

But running `sync(drafts)` concurrently can cause a problem here. All running processes would depend on the very same instance of a service, but an access to the cache isn't synchronized. 
This applies also to other services.